Civi/Civioffice/Render/Queue/RenderQueueRunner.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/*
+ * Copyright (C) 2024 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types = 1);
+
+namespace Civi\Civioffice\Render\Queue;
+
+use Civi\Civioffice\CiviofficeSession;
+use CRM_Civioffice_ExtensionUtil as E;
+
+class RenderQueueRunner {
```

Comment:
Any specific architectural reason why this is not a subclass of `CRM_Queue_Runner`?

Reply:
`CRM_Queue_Runner` not only runs a queue, but also contains the queue itself. This one runs the jobs given in a `RenderQueue` as method argument. Therefore it can exist without a queue.

Reply:
Sure, but then again, I wonder why the `RenderQueue` is not a member of the `renderQueueRunner`, since the runner is useless without a queue, and the `RenderQueue` might be a subclass of `CRM_Queue_Queue`. So the question would by why you chose to not subclass them and make them independent classes with Core queue objects as members …?

Reply:
The methods of the `CRM` classes are not required here, so there's no reason to offer them which simplifies the usage. Because the `RenderQueueRunner` is instantiable without queue it's possible to inject it as service and thus allows to create *unit*-testable code. Also it would be easier to change the current used `CRM` classes with something else. I think applying the delegation over composition principle makes sense here.